md/1021.md

```diff
@@ -0,0 +1,96 @@
+โจทย์ข้อนี้มีคำสั่ง $N$ คำสั่ง โดยคำสั่งจะมีสองแบบ
+
+1. ใส่สินค้ามูลค่า $i$ เข้าไปในเครื่อง
+2. ถามว่าสินค้าที่มีมูลค่ามากสุดในเครื่องมีมูลค่าเท่าใดหากมีและเอาออกจากเครื่อง
+
+โจทย์ข้อนี้แก้ได้ด้วย Priority Queue ซึ่งเป็นโครงสร้างข้อมูลที่มีประสิทธิภาพในการหาค่าสูงสุดของชุดข้อมูลที่อาจมีการเพิ่มเข้าหรือลบออก
+
+ทั้งนี้ Priority Queue เป็นโครงสร้างข้อมูลที่มีอยู่ใน STL ซึ่งทำให้ข้อนี้แก้ได้ง่ายมากหากใช้ std::priority_queue โดยเพียงต้องใช้ฟังก์ชัน push เพื่อคำสั่งประเภทแรก และ top กับ pop สำหรับคำสั่งประเภทสอง
```

Comment:
```suggestion
ทั้งนี้ Priority Queue เป็นโครงสร้างข้อมูลที่มีอยู่ใน STL ซึ่งทำให้ข้อนี้แก้ได้ง่ายมากหากใช้ `std::priority_queue` โดยเพียงต้องใช้ฟังก์ชัน `push` เพื่อคำสั่งประเภทแรก และ `top` กับ `pop` สำหรับคำสั่งประเภทสอง
```

Reply:
![image](https://github.com/programming-in-th/solutions/assets/26710819/b1075872-c697-4b2c-a8f5-9bfe7a49d183)

Reply:
Our solution pages are malfunctioning at this moment. We are trying to fix it.